### PR TITLE
Replaced explicit variable types with var/final var

### PR DESCRIPTION
**Pull request by MarcinOrlowski**
Replaced explicit variable types with var/final var. Small code cleanup + change to `fireNodeChanged()` in src/main/java/com/cburch/logisim/gui/main/SimulationTreeModel.java and no other logic changes

### CODE REVIEW

**Review by davidhutchens — APPROVED** [review]
This looks good to me.

If 3.9 is going to be released soon, you might consider holding this PR until after that release since it provides no new features nor corrects errors, but it does clean up some logic and thus has a very small chance of introducing a problem.

**Review by maehne — APPROVED** [review]

**Comment by MarcinOrlowski** [comment]
@davidhutchens I can roll any changes back if that would help, no problem. But before doing that, I'd ask another review round please? These non "final var" type of changes (and I have literally a few of them) are usually reduction of LoC by replacing old style code with separate declaration and then conditional assignment with just single ternary final assignment. I intentionally do not do anything more complex right now for the sake of review. Please let me know.

**Comment by davidhutchens** [comment]
@MarcinOrlowski: I approved the changes after careful review. I am pretty sure you did not introduce a problem. I made the additional comment because we are presumably getting close to a new release and should be cautious of changes. I certainly did not intend to slow you down in your PRs, nor did I expect you to reverse any changes in this one. I'm just not sure what @BFH-ktt1 is thinking on the timing of the next release. The "you" in my comment was a poor choice. I was intending to direct the comment to @BFH-ktt1.

**Pull request merged**
